apps/golden-sample-app/src/app/locale-selector/locale-selector.component.ts

```diff
@@ -38,6 +43,8 @@ export class LocaleSelectorComponent implements OnInit {
 
   ngOnInit() {
     this.currentLanguage = this.findLocale(this.localeService.currentLocale);
+    const event = new CustomEvent('data-analytics-change-language', { detail: { locale: this.currentLanguage } });
+    window.dispatchEvent(event);
```

Comment:
This is my favourite way of doing it, perhaps dispatching on document instead of window makes it a little bit easier to inject a value in Angular (I know it's not needed but strict TS validation might throw errors)